### PR TITLE
feat: adds initial cut at datadog_synthetics_test resource

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Adds support for creating DataDog Synthetic Tests

## why
* Synthetic Tests are very similar to monitors and they end up creating a monitor under the hood, so it's useful to create them via the same mechanism / module.

### CODE REVIEW

**Comment by osterman** [comment]
## Question
Do the alert tags really need to be separated this way? E.g. what about just invoking the module twice with different alert tags if a user really needed that? That gives them the ultimate flexibility of slicing/dicing the tags how ever the want, rather than putting it into the module.

**Comment by Gowiem** [comment]
@osterman good point -- I had a use-case for it on a project obviously, but if we want to be tidy here then I take your approach. I'll reset and push the earlier version.

**Review by osterman — APPROVED** [review]
LGTM, but better @danjbh or @danjbh review the PR.

File: variables.tf [hunk 1]
```diff
@@ -22,7 +22,27 @@ variable "datadog_monitors" {
...
+    status    = string
+    locations = list(string)
+    tags      = list(string)
+    assertions = list(object({
+      type     = string
+      operator = string
+      target   = any
+    }))
+  }))
+  description = "List of DataDog synthetic test configurations. See catalog for examples"
```
Comment (aknysh) [inline variables.tf:45]:
```suggestion
  description = "List of Datadog synthetic test configurations. See catalog for examples"
```

Comment (Gowiem) [reply variables.tf:45]: Ha, I've always written it this way... but you're right 👍

**Comment by aknysh** [comment]
> ## Question
> Do the alert tags really need to be separated this way? E.g. what about just invoking the module twice with different alert tags if a user really needed that? That gives them the ultimate flexibility of slicing/dicing the tags how ever the want, rather than putting it into the module.

Agree on `alert_tags`, we can invoke the module many times for diff tags and diff monitors/synthetics

**Comment by osterman** [comment]
image

**Comment by Gowiem** [comment]
Updated this PR to also enforce v0.13 to address #22 (I've seen that in a previous PR and it is why I needed to add the `terraform/0.13` label). 

Closes #22

**Comment by Gowiem** [comment]
Hey @aknysh -- Question for you if you don't mind as I believe you're the SME on the yaml-config module. I'm having a problem with this module + using the `terraform-yaml-config` module. I'm getting an esoteric error when adding multiple maps/objects to one of the yaml config file: 

image

You should be able to see this error in the current test failures. 

If I remove the `simple` map or if I remove the `complex` map from [this file](https://github.com/cloudposse/terraform-datadog-monitor/pull/25/files#diff-9e6f72721707edc6d03e8d792177ac9bd9a02fb4e1f21d36b4aacee6e1aacbb0R1) then everything plans correctly. But if there are two elements in the resulting map from that file then it fails with the above `collection elements cannot be unified` error. Have you seen this before? It seems quite odd and I'm wondering if I'm missing something small in my yaml config. Any thoughts?

**Comment by Gowiem** [comment]
@aknysh apologies, you can ignore my question. My issue was due to Terraform's `map` type not being able to handle a map object that is sometimes flat and other types 1 level deep. I fixed that in my latest commit by removing any ambiguity with the input var.

File: examples/complete/main.tf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module "yaml_config" {
+module "monitor_configs" {
   source = "git::https://github.com/cloudposse/terraform-yaml-config.git?ref=tags/0.1.0"
```
Comment (danjbh) [inline examples/complete/main.tf:2]: Can we migrate these source references to this new format?

```suggestion
  source  = "cloudposse/yaml/config"
  version = "0.1.0"
```

File: examples/complete/main.tf [hunk 3]
```diff
@@ -7,10 +7,20 @@ module "yaml_config" {
   context = module.this.context
 }
 
+module "synthetic_configs" {
+  source = "git::https://github.com/cloudposse/terraform-yaml-config.git?ref=tags/0.1.0"
```
Comment (danjbh) [inline examples/complete/main.tf:11]: See previous note about migrating to new source reference format

**Review by danjbh — APPROVED** [review]
Just a small nitpicky thing, otherwise it looks good to me!

**Comment by Gowiem** [comment]
@danjbh good call out -- fixed!

**Review by danjbh — APPROVED** [review]

**Pull request merged**